langchain4j-core/src/main/java/dev/langchain4j/model/chat/request/ChatRequest.java

```diff
@@ -3,26 +3,89 @@
 import dev.langchain4j.Experimental;
 import dev.langchain4j.agent.tool.ToolSpecification;
 import dev.langchain4j.data.message.ChatMessage;
+import dev.langchain4j.model.chat.request.json.JsonSchema;
 
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Objects;
 
 import static dev.langchain4j.internal.Utils.copyIfNotNull;
+import static dev.langchain4j.internal.Utils.quoted;
 import static dev.langchain4j.internal.ValidationUtils.ensureNotEmpty;
+import static dev.langchain4j.model.chat.request.ResponseFormatType.JSON;
 import static java.util.Arrays.asList;
 
 @Experimental
 public class ChatRequest {
```

Comment:
Couldn't this become a `record`?

Reply:
Good point, will do

Reply:
I was also thinking if `ChatRequest` should be an interface.

Use case: LC4j user keeps some/all of the parameters (e.g., `modelName`, `temperature`, `maxOutputTokens`, etc) in some external system (e.g., DB) and does not want to map all properties manually from their model class into `ChatRequest`.

But now after writing this thought down I feel like I am overthinking it. WDYT?

Reply:
Yes, `ChatRequest` could have a `DefaultChatRequest` implementation which will be created by the builder.

Let's suppose user wants to keep track of these parameters in the DB table. Perharps each user of their system want's to customize the parameters of the model.

```java
@Entity
public class ChatRequestConfig implements ChatRequest {

    private final String modelName;
    private final Double temperature;
    private final Integer maxOutputTokens;
    ...

    @Override
    public String modelName() { // this is a method declared in the ChatRequest
        return modelName;
    }

    ...
}
```

In this way they could make their model classes implement `ChatRequest` and supply it into `ChatModel.chat()` method directly, otherwise they will have to do a mapping for every parameter:

```java
ChatRequestConfig config = getChatRequestConfig();

ChatRequest chatRequest = ChatRequest.builder()
    .modelName(config.modelName())
    .temperature(config.temperature())
    ...
    build();

chatModel.chat(chatRequest)

```

Reply:
That's a bit stretched, but I see the point, so probably you could go for it. There is only one more point to consider: if you turn the `ChatRequest` into an interface and expect users to eventually provide their own implementation, then you're making this pattern part of your public API, meaning that you won't be allowed to add more (non-default) methods to that interface (or change existing ones) without breaking backward compatibility. 

Note that I'm not saying that you shouldn't do that, I'm only brainstorming to weight pros and cons.

Reply:
Hmm, good point. I do not expect many users will use their own implementation.
I expect to add more fields to `ChatRequest` though. This will mean that the default implementation for accessors of newly added fields will have to return null (or some sane default, if applicable), which should work fine in theory?

Reply:
Right, probably this won't be a big issue, just pointing it out.

Reply:
It seems it is time to apply IIDLIO principle here :)
I will probably keep it as-is for now and convert it into an interface later if/when it will be required.

Reply:
@glaforge I guess so, although every LLM provider implements caching differently, so we shall see